snap/info.go

```diff
@@ -538,6 +559,8 @@ func ReadInfo(name string, si *SideInfo) (*Info, error) {
 		return nil, err
 	}
 
+	info.renameClashingCorePlugs()
```

Comment:
Having this in two places makes me wonder if we should move it instead into `InfoFromSnapYaml()` ?